scripts/wikitables_eval.py

```diff
@@ -0,0 +1,98 @@
+                    level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+sys.path.insert(0, os.path.dirname(os.path.abspath(os.path.join(__file__, os.pardir))))
+from allennlp.common.file_utils import cached_path
+from allennlp.data import DatasetReader
+from allennlp.data.iterators import BasicIterator
+from allennlp.models.archival import load_archive
+
+DEFAULT_EXECUTOR_JAR = "https://s3-us-west-2.amazonaws.com/allennlp/misc/wikitables-executor-0.1.0.jar"
+ABBREVIATIONS_FILE = "https://s3-us-west-2.amazonaws.com/allennlp/misc/wikitables-abbreviations.tsv"
+GROW_FILE = "https://s3-us-west-2.amazonaws.com/allennlp/misc/wikitables-grow.grammar"
+
+
+def main(args: argparse.Namespace):
+    predict_logical_forms(args)
+    evaluate_logical_forms(args)
+
+
+def predict_logical_forms(args: argparse.Namespace):
+    os.makedirs(args.output_dir, exist_ok=True)
+    archive = load_archive(args.model_archive, cuda_device=args.cuda_device)
+    model = archive.model
+    model.eval()
+    # TODO(mattg): TOTAL HACK! Not sure why I'm getting this - might be due to numbers?
+    model.vocab._token_to_index['rule_labels']['@@UNKNOWN@@'] = 0
+    config = archive.config
+    config['dataset_reader']['lazy'] = True
+    dataset_reader = DatasetReader.from_params(config['dataset_reader'])
+    dataset = dataset_reader.read(args.test_data)
+    data_iterator = BasicIterator(args.batch_size)
+    data_iterator.index_with(model.vocab)
+    num_batches = data_iterator.get_num_batches(dataset)
+    logical_forms = []
+    for batch in tqdm.tqdm(data_iterator(dataset, num_epochs=1, shuffle=False)):
+        if 'target_action_sequences' in batch:
+            # This makes the model skip the loss computation, which will make things a bit faster.
+            del batch['target_action_sequences']
+        results = model(**batch)
+        logical_forms.extend(results['logical_form'])
+    logical_form_filename = os.path.join(args.output_dir, 'logical_forms.txt')
+    with open(logical_form_filename, 'w') as logical_form_file:
+        for logical_form in logical_forms:
+            logical_form_file.write(f"{logical_form}\n")
+
+
+def evaluate_logical_forms(args: argparse.Namespace):
+    logical_form_filename = os.path.join(args.output_dir, 'logical_forms.txt')
+    # Sempre relies on having a few files available in data/ in the current directory.  It's easier
+    # to just download them here instead of trying to change the sempre code...
+    keep_data_dir = False
+    if os.path.exists('data'):
+        if not os.path.isdir('data'):
+            raise RuntimeError("A file named 'data' exists in the current directory; can't proceed")
+        else:
+            keep_data_dir = True
+    os.makedirs('data', exist_ok=True)
+    if not os.path.exists('data/abbreviations.tsv'):
+        run(f'wget {ABBREVIATIONS_FILE}', shell=True)
+        run('mv wikitables-abbreviations.tsv data/abbreviations.tsv', shell=True)
```

Comment:
Looks like you're using allenai's fork of sempre (the original one doesn't need this file).
How hard is it to pull the latest code from Percy's fork and build a jar automatically in this script?

Reply:
Never mind, pulling the latest changes may not be a great idea because we would want the executor fixed.

Reply:
I was using the code that PNP is using.  I'd be very wary of pulling dependencies and building a jar in the script, as that'd be very unstable.